src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java

```diff
@@ -490,6 +505,44 @@ public void getResult_asStudent() throws Exception {
         request.get("/api/results/" + result.getId(), HttpStatus.FORBIDDEN, Result.class);
     }
 
+    @Test
+    @WithMockUser(value = "instructor1", roles = "INSTRUCTOR")
+    public void testGetResultsForExamExercise() throws Exception {
+        var now = ZonedDateTime.now();
+        for (int i = 1; i <= 10; i++) {
+            ModelingSubmission modelingSubmission = new ModelingSubmission();
+            modelingSubmission.model("Text");
+            modelingSubmission.submitted(true);
+            modelingSubmission.submissionDate(now.minusHours(3));
+            database.addSubmission(this.examModelingExercise, modelingSubmission, "student" + i);
+            if (i % 3 == 0) {
+                database.addResultToSubmission(modelingSubmission, AssessmentType.MANUAL, database.getUserByLogin("instructor1"), 10D, true);
+            }
+            else if (i % 4 == 0) {
+                database.addResultToSubmission(modelingSubmission, AssessmentType.SEMI_AUTOMATIC, database.getUserByLogin("instructor1"), 20D, true);
+            }
+        }
+
+        // empty participation with submission
+        StudentParticipation participation = new StudentParticipation();
+        participation.setInitializationDate(ZonedDateTime.now());
+        participation.setParticipant(null);
+        participation.setExercise(this.examModelingExercise);
+        studentParticipationRepository.save(participation);
+        ModelingSubmission modelingSubmission = new ModelingSubmission();
+        modelingSubmission.model("Text");
+        modelingSubmission.submitted(true);
+        modelingSubmission.submissionDate(now.minusHours(3));
+        participation.addSubmission(modelingSubmission);
+        modelingSubmission.setParticipation(participation);
+        submissionRepository.save(modelingSubmission);
+        studentParticipationRepository.save(participation);
+
+        List<Result> results = request.getList("/api/exercises/" + this.examModelingExercise.getId() + "/results", HttpStatus.OK, Result.class);
+        assertThat(results).hasSize(5);
+        // TODO: check additional values
```

Comment:
Am I a bit premature or did you forget to remove this comment?

Reply:
yeah I will remove it :D